File: orderservice/orderservice/src/main/java/com/e_commerce_project/orderservice/Service/OrderService.java
package com.e_commerce_project.orderservice.Service;

import com.e_commerce_project.orderservice.Records.OrderRequest;
import com.e_commerce_project.orderservice.Records.OrderResponse;

import java.util.List;

public interface OrderService {
    Long createOrder(OrderRequest orderRequest);

    List<OrderResponse> getAllOrders();

    OrderResponse getOrderById(Long id);
}

File: e-commerce-frontend/src/app/services/order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { OrderRequest, OrderResponse } from '../models/order.model';

@Injectable({
    providedIn: 'root'
})
export class OrderService {
    private apiUrl = environment.apiUrls.order;

    constructor(private http: HttpClient) { }

    /**
     * Create a new order
     */
    createOrder(order: OrderRequest): Observable<number> {
        return this.http.post<number>(`${this.apiUrl}/createOrder`, order);
    }

    /**
     * Get all orders
     */
    getAllOrders(): Observable<OrderResponse[]> {
        return this.http.get<OrderResponse[]>(`${this.apiUrl}`);
    }

    /**
     * Get order by ID
     */
    getOrderById(id: number): Observable<OrderResponse> {
        return this.http.get<OrderResponse>(`${this.apiUrl}/${id}`);
    }
}
